pkg/devserver/server.go

```diff
@@ -31,18 +37,101 @@ func (s *Server) Start() error {
 	return nil
 }
 
+type PageTemplate struct {
+	Title   string
+	Content string
+}
+
+var pageTemplateContent = `<!DOCTYPE html>
+<html>
+	<head>
+		<title>{{ .Title }}</title>
+		<link rel="stylesheet" href="/assets/css/main.css">
+	</head>
+	<body>
+		<h1>{{ .Title }}</h1>
+		{{ .Content }}
+	</body>
+</html>`
+
+var pageTemplate *template.Template
+
+type PageData struct {
+	Title   string
+	Content template.HTML
+}
+
+func init() {
+	t, err := template.New("page").Parse(pageTemplateContent)
+	if err != nil {
+		panic(err)
+	}
+	pageTemplate = t
+}
+
+type Renderer struct{}
+
+func (r *Renderer) Render(w io.Writer, name string, data interface{}, c echo.Context) error {
+	if name != "page" {
+		return fmt.Errorf("unknown template: %s", name)
+	}
+
+	return pageTemplate.Execute(w, data)
+}
+
+func serveNotFound(c echo.Context) error {
+	return c.Render(http.StatusNotFound, "page", PageData{
+		Title:   "Not found!",
+		Content: template.HTML("<p>Looks like this page doesn't exist yet</p>"),
+	})
+}
+
 func (s *Server) servePage(c echo.Context) error {
-	file, err := content.ParseFile(path.Join(s.config.ContentDir, c.Param("page")+".md"))
+	page := c.Param("page")
+	contentPath := path.Join(s.config.ContentDir, page+".md")
+
+	_, err := os.Stat(contentPath)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return serveNotFound(c)
+		}
+		return fmt.Errorf("error checking file: %w", err)
+	}
+
+	file, err := content.ParseFile(contentPath)
 	if err != nil {
 		return fmt.Errorf("error processing file: %w", err)
 	}
 
-	return c.HTMLBlob(http.StatusOK, file.ParsedContent)
+	return c.Render(http.StatusOK, "page", PageData{
+		Title:   page,
+		Content: template.HTML(string(file.ParsedContent)),
+	})
 }
 
+//go:embed static
+var staticFS embed.FS
+
 func NewServer(config Config) *Server {
 	echoInst := echo.New()
 
+	var configuredFrontendFS http.FileSystem
+	if config.UseBundledAssets {
+		configuredFrontendFS = http.FS(staticFS)
```

Comment:
For me, using bundled assets is giving me a 404. This can be fixed by doing one of:
- Remove the intermediary `static` folder and just have `assets/css/main.css`
- Change the `http.Dir` to remove the `/static/`, and update all references to the static assets to include `/static` in front
- Replace the direct reference here to `staticFs` to a sub-filesystem trimming the leading `static` from paths. You can get this by adding `assetsFs, _ := fs.Sub(staticFS, "static")`

Reply:
weird, unsure why the setup on my end would work and yours would not, will look into this

Reply:
fixed in https://github.com/fogo-sh/almanac/pull/2 , looks like i didn't even bother testing it when i made this commit until the other branch